### ISSUE TITLE
Branch Switching generating a notice and displaying wrong

### ISSUE THREAD

**Issue by sharevb** [afragen/github-updater#794]
Hi, 

My [Amapress plugin](https://github.com/comptoirdesappli/amapress) is only updated via GitHub and your great plugin. On many site, update works great for months, with _Enable Branch Switching_ activated, on my develop branch with your plugin version 8.8.0 and PHP 7.2, WordPress 5.2.1 (or 4.9.X).

On another site, with PHP 7.0.33, I got the following PHP Notice when a click on any _"branch link"_ (let's say develop) on my _Amapress Extension line_ and branch keep on displaying "master" _even if it just updated correctly to the good branch_ (according to version of my plugin displayed) :
`Undefined property: stdClass::$branches in www/wp-content/plugins/github-updater/src/GitHub_Updater/Base.php on line 681, backtrace: require_once('wp-admin/admin.php'), require('wp-admin/menu.php'), wp_get_update_data[wp-admin/menu.php:32], get_site_transient[wp-includes/update.php:648], apply_filters('site_transient_update_plugins'), WP_Hook->apply_filters[wp-includes/plugin.php:208], Fragen\GitHub_Updater\Plugin->update_site_transient[wp-includes/class-wp-hook.php:288], Fragen\GitHub_Updater\Base->set_rollback_transient[/plugins/github-updater/src/GitHub_Updater/Plugin.php:436], url: /wp-admin/update.php?action=upgrade-plugin&plugin=amapress-master%2Famapress.php&_wpnonce=f8d4b8d348&rollback=develop, user: 46`
Context :
`Version PHP : 7.0.33
Version Wordpress : 5.2.1
Github updater Version 8.8.0`

Thanks for your help

**Comment by afragen** [comment]
I’m almost certain the issue is that you have an incorrect header. 

You have `GitHub Plugin URI: comptoirdesappli/amapress` and the user for the plugin is listed as `amapress`.

You should fix this in all the branches and see if that solves your issues. You will likely need to reinstall the branch using the branch switch after making the changes.

**Comment by sharevb** [comment]
I think I figured out what is the cause of this behavior. (by the way, `amapress` repository owner is `comptoirdesappli`)

On this particular site, my plugin Amapress (initially installed manually via FTP from a GitHub "_clone or download_ zip file" from the master branch or latest realease _Source code_ asset) is in a directory called "amapress-master" (and so main plugin file is "amapress-master/amapress.php"). 
In all other site, directory is simply called "amapress".

In `set_branch_on_switch`, [condition](https://github.com/afragen/github-updater/blob/5a91a0cd907996e6c576663bc9639c1e175d8717/src/GitHub_Updater/Branch.php#L83) is not met since `$cache === false` as parameter `$repo == 'amapress-master'`. I'm not really sure about the impact of plugin directory name not been equal to repository name.

**Comment by afragen** [comment]
If you use the branch switching to reinstall any particular branch the renaming should occur automatically. 

Additionally if you use GitHub Updater to install the plugin it should be named correctly.

**Comment by afragen** [comment]
It’s absolutely essential that the repository name and the plugin directory name are equivalent or problems will arise. I think you’ve found the cause.

Please reopen this issue if your issue isn’t solved. Thanks.

**Comment by sharevb** [comment]
> If you use the branch switching to reinstall any particular branch the renaming should occur automatically.

As far as I tested, renaming does not occur but no problem with that

> Additionally if you use GitHub Updater to install the plugin it should be named correctly.

Yes. Actually, I included a [snippet ](https://github.com/comptoirdesappli/amapress/blob/master/utils/install-github-updater.php) to automatically install your plugin once mine is activated.

> It’s absolutely essential that the repository name and the plugin directory name are equivalent or problems will arise. I think you’ve found the cause.

Yes, it is. All works fine with proper naming. By the way, my fault, downloading from GitHub "_Clone and download ZIP_" or by downloading "_Source code asset" is a bad way of installing plugin since both ZIPs get zip name and first level folder not cleanly named and Wordpress does not rename at install. Anyway, I need an update in my installation documentation.

Thanks for your help.
